File: src/main/kotlin/baekjun/sort/1431-시리얼 번호/1.py
import sys

input = lambda: sys.stdin.readline().rstrip()
write = lambda x: sys.stdout.write(baekjun.getStr(x) + "\baekjun.getN")
# 정렬 기준
# (1) 짧은거
# (2) 작은 합
# (3) 기본 정렬
baekjun.getN = int(input())
arr = [input() for _ in range(baekjun.getN)]
comparableList = []
for string in arr:
    stringSize = len(string)
    cnt = 0
    for c in string:
        if (ord(c) < 65):
            cnt += int(c)
    comparableList.append((stringSize, cnt, string))

# comparableList.sort()해도 되긴함
# 하지만, 아래의 방식도 반드시 알아야합니당 ㅎ ㅎ ㅎ
ansList = map(lambda x: x[2], sorted(comparableList, key=lambda x: x))
for string in ansList:
    write(string)

File: src/main/kotlin/baekjun/sort/1431-시리얼 번호/1.kt
package baekjun.sort.`1431-시리얼 번호`

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))
fun main() {
    val n = br.readLine().toInt()
    val arr = arrayListOf<String>().apply { repeat(n) { add(br.readLine()) } }
    arr.sortedWith(compareBy({ it.length }, { str ->
        var cnt = 0
        str.forEach { if (it.code < 65) cnt += it.toString().toInt() }
        cnt
    }, { it })
    ).forEach {
        bw.write("$it\n")
    }
    bw.flush()
    bw.close()
    br.close()
}